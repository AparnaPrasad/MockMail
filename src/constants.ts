export enum FolderTypes {
    T_INBOX="Inbox",
    T_DRAFTS="Drafts",
    T_SENT="Sent",
    T_SPAM="Spam",
    T_TRASH="Trash"
}

export const FolderList: FolderTypes[] = [FolderTypes.T_INBOX, FolderTypes.T_DRAFTS, 
    FolderTypes.T_SENT,
    FolderTypes.T_SPAM,
    FolderTypes.T_TRASH ]

export const baseUrl = 'http://localhost:8000'