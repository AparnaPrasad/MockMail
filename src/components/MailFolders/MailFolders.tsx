import React from 'react';
import { FolderTypes, FolderList } from '../../constants';
import { Button } from 'react-bootstrap';
import { useMapState } from '../../MapProvider';
import { ActionTypes } from '../../MapActions';
const styles = {
    listElement:{
        listStyle: 'none'
    },
    selected: {
        background: 'red'
    }
}



const MailFolders: React.FC = () => {
    const { mapState: { selectedFolder },
        setMapState } = useMapState();
    const selectFolder = (prevSelectedFolder: FolderTypes, newFolder: FolderTypes) => {
        if (prevSelectedFolder !== newFolder) {
            setMapState({ type: ActionTypes.SET_SELECTED_FOLDER, selectedFolder: newFolder })
        }
    }
    const getStyle = (selectedFolder: FolderTypes, folder: FolderTypes) => {
        return selectedFolder === folder ? { ...styles.listElement, ...styles.selected } : { ...styles.listElement }
    }
    return (<React.Fragment>
        <div>Mail</div> 
        <Button>Compose</Button>
          <ul>
            {FolderList.map((folder) => (<li key={folder}
                style={getStyle(selectedFolder, folder)}
                onClick={() => { selectFolder(selectedFolder, folder) }}>{folder}</li>))}
          </ul>
    </React.Fragment> );
}

export default MailFolders;
