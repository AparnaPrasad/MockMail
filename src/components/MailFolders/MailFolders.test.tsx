import React from 'react';
import MailFolders from './MailFolders';
import { shallow } from 'enzyme';
import { FolderList } from '../../constants';

describe(('renders side bar correctly'), () => {

    it('should render list of folders', () => {
        //const folderList =
        const wrapper = shallow(<ul>
            {FolderList.map((folder) => (<li key={folder}>{folder}</li>))}

        </ul>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should have a compose button', () => {

        const wrapper = shallow(<MailFolders/>)
        //There should be only one button
        expect(wrapper.find('Button')).toHaveLength(1);

        

        //Button should have matching text
        expect(wrapper.find('Button').text()).toEqual('Compose');
    });
})

