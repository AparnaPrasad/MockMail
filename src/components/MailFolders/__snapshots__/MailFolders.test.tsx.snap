// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders side bar correctly should render list of folders 1`] = `ShallowWrapper {}`;
